app/controllers/concerns/json_api_pagination.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module JsonApiPagination
+  extend ActiveSupport::Concern
```

Comment:
I always find cconcerns concerning. I'd move this to a service with a clear API. This way there's just a random use of the "parent" properties (pagy and params) with no clear boundaries. In this case, the ControllerPaginationService would receive pagy itself, params[:per_page], and params[:page].